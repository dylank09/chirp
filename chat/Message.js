import React from "react";
import { StyleSheet, Text, View } from "react-native";

import { theme } from "../assets/Theme";

import firebase from "firebase/app";

export default function Message({ text, timestamp, user, me }) {
  return (
    <View
      style={{
        flexDirection: "row",
        minWidth: "50",
        borderRadius: 18,
        padding: 5,
        marginLeft: 5,
        marginRight: 5,
        marginTop: 3,
        height: theme.dimensions.inputHeight + 5,
        backgroundColor: me ? theme.colors.primary : theme.colors.secondary,
      }}
    >
      <Text style={styles.user}>{user ? user : "U"}</Text>
      <Text style={styles.text}>{text ? text : "Sample text"}</Text>
      <Text style={styles.ts}>{timestamp ? timestamp : "16:01"} </Text>
    </View>
  );
}

const styles = StyleSheet.create({
  user: {
    padding: 4,
    alignSelf: "flex-start",
  },
  text: {
    padding: 4,
    alignSelf: "flex-end",
    fontSize: theme.dimensions.standardFontSize + 3,
  },
  ts: {
    padding: 2,
  },
});
