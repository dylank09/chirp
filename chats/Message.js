import React from "react";
import { StyleSheet, Text, View } from "react-native";

import { theme } from "../assets/Theme";
import FormatTime from "../functions/FormatTime";

export default function Message({ text, timestamp, user, me }) {
  timestamp = FormatTime(timestamp);

  if (text.length < 15 && user.length > 5) {
    user = user.substring(0, user.indexOf(" "));
  }

  return (
    <View style={styles.row}>
      <View
        style={[
          styles.messageBox,
          {
            alignSelf: me ? "flex-end" : "flex-start",
            backgroundColor: me ? theme.colors.primary : theme.colors.secondary,
            textAlign: me ? "right" : "left",
          },
        ]}
      >
        <View style={styles.userAndTime}>
          <Text style={styles.user}>{user ? user : ""}</Text>
          <Text style={styles.ts}>{timestamp} </Text>
        </View>
        <Text style={[styles.text, { textAlign: me ? "right" : "left" }]}>
          {text ? text : ""}
        </Text>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  row: {
    width: "100%",
    alignContent: "flex-start",
    marginBottom: 5,
  },
  messageBox: {
    borderRadius: theme.dimensions.inputHeight / 2,
    flexDirection: "column",
    maxWidth: "62%",
    padding: 6,
    marginHorizontal: 4,
  },
  userAndTime: {
    flexDirection: "row",
    marginLeft: 4,
    justifyContent: "space-between",
  },
  user: {
    color: theme.colors.jet,
    alignSelf: "flex-start",
    marginRight: 2,
    fontWeight: "700",
    fontSize: theme.dimensions.standardFontSize - 4,
  },
  ts: {
    alignSelf: "flex-start",
    marginLeft: 2,
    color: theme.colors.jet,
    fontWeight: "700",
    fontSize: theme.dimensions.standardFontSize - 4,
  },
  text: {
    width: "auto",
    color: theme.colors.text,
    padding: 4,
    fontSize: theme.dimensions.standardFontSize,
  },
});
