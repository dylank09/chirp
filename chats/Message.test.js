import React from "react";
import renderer from "react-test-renderer";
import { render } from "@testing-library/react-native";

import Message from "./Message";

describe("<Message />", () => {
  const tree = renderer.create(<Message />).toJSON();

  it("renders correctly", () => {
    expect(tree).toMatchSnapshot();
  });

  it("user is present", () => {
    const { getByTestId } = render(<Message user="Testing User" />);
    const foundUser = getByTestId("user");
    expect(foundUser).toBeTruthy();
  });

  it("time is present", () => {
    const { getByTestId } = render(<Message timestamp={1613748319} />);
    const foundTime = getByTestId("time");
    expect(foundTime).toBeTruthy();
  });

  it("text is present", () => {
    const { getByTestId } = render(<Message text="Testing message" />);
    const foundText = getByTestId("message");
    expect(foundText).toBeTruthy();
  });
});
