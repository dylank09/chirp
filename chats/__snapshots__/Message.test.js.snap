// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Message /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignContent": "flex-start",
      "marginBottom": 5,
      "width": "100%",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "borderRadius": 25,
          "flexDirection": "column",
          "marginHorizontal": 4,
          "maxWidth": "62%",
          "padding": 6,
        },
        Object {
          "alignSelf": "flex-start",
          "backgroundColor": "rgb(132, 135, 165)",
          "textAlign": "left",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginLeft": 4,
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "flex-start",
            "color": "rgb(45, 45, 42)",
            "fontSize": 11,
            "fontWeight": "700",
            "marginRight": 2,
          }
        }
        testID="user"
      >
        
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "flex-start",
            "color": "rgb(45, 45, 42)",
            "fontSize": 11,
            "fontWeight": "700",
            "marginLeft": 2,
          }
        }
        testID="time"
      >
        
         
      </Text>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "rgb(255, 255, 255)",
            "fontSize": 15,
            "padding": 4,
            "width": "auto",
          },
          Object {
            "textAlign": "center",
          },
        ]
      }
      testID="message"
    >
      
    </Text>
  </View>
</View>
`;
